lib/Raven/Client.php

```diff
@@ -747,32 +735,10 @@ public function send(&$data)
             return;
         }
 
-        $message = $this->encode($data);
-
-        $headers = [
-            'User-Agent' => static::getUserAgent(),
-            'X-Sentry-Auth' => $this->getAuthHeader(),
-            'Content-Type' => 'application/octet-stream'
-        ];
-
-        $config = $this->getConfig();
-        $server = sprintf('%s/api/%d/store/', $config->getServer(), $config->getProjectId());
-
-        $this->send_remote($server, $message, $headers);
-    }
+        $uri = sprintf('api/%d/store/', $this->getConfig()->getProjectId());
+        $request = $this->requestFactory->createRequest('POST', $uri, [], JSON::encode($data));
 
-    /**
-     * Send data to Sentry
-     *
-     * @param string       $url     Full URL to Sentry
-     * @param array|string $data    Associative array of data to log
-     * @param array        $headers Associative array of headers
-     */
-    protected function send_remote($url, $data, $headers = [])
-    {
-        $parts = parse_url($url);
-        $parts['netloc'] = $parts['host'].(isset($parts['port']) ? ':'.$parts['port'] : null);
-        $this->send_http($url, $data, $headers);
+        $this->httpClient->sendAsyncRequest($request)->wait(true);
```

Comment:
Hmmm yes this is true. Depending on how hard we want to push on reading a potential 429 or error response I would say go for async since it's faster. 

Going async allows us to have a callback right? We could possibly read and honor a 429 there, or an error response.

Given the nature of php being per request a new world, honoring those things is a bit annoying anyhow.